openshift/oc_promote.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+#%
+#% OpenShift Deployment Promotion Helper
+#%
+#%   Intended for use with a pull request-based pipeline.
+#%
+#% Usage:
+#%
+#%   ${THIS_FILE} [PR_NUMBER] [apply]
+#%
+#% Examples:
+#%
+#%   Provide a PR number. Defaults to a dry-run.
+#%   ${THIS_FILE} 0
+#%
+#%   Apply when satisfied.
+#%   ${THIS_FILE} 0 apply
+#%
+#
+# If no parameters, then show this help header (cat file, grep #% lines and clean up with sed)
+#
+[ "${#}" -gt 0 ] || {
+	THIS_FILE="$(dirname ${0})/$(basename ${0})"
+	cat ${THIS_FILE} |
+		grep "^#%" |
+		sed -e "s|^#%||g" |
+		sed -e "s|\${THIS_FILE}|${THIS_FILE}|g"
+	exit
+}
+
+# Specify halt conditions (errors, unsets, non-zero pipes), field separator and verbosity
+#
+set -euo pipefail
+IFS=$'\n\t'
+[ ! "${VERBOSE:-}" == "true" ] || set -x
```

Comment:
Same here it's hard to read the intended logic
how about [ "${VERBOSE:-}" == "true" ] && set -x